File: jscomp/test/gpr_1822_test.js
'use strict';

var Mt = require("./mt.js");
var Block = require("../../lib/js/block.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  test_id.contents = test_id.contents + 1 | 0;
  suites.contents = /* :: */{
    _0: /* tuple */[
      loc + (" id " + String(test_id.contents)),
      (function (param) {
          return {
                  tag: /* Eq */0,
                  _0: x,
                  _1: y
                };
        })
    ],
    _1: suites.contents
  };
  
}

var myShape = {
  tag: /* Circle */0,
  _0: 10
};

var area;

area = myShape.tag ? Math.imul(10, myShape._1) : 100 * 3.14;

eq("File \"gpr_1822_test.ml\", line 21, characters 6-13", area, 314);

Mt.from_pair_suites("Gpr_1822_test", suites.contents);

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.myShape = myShape;
exports.area = area;
/* area Not a pure module */

File: jscomp/test/gpr_1822_test.ml
let suites : Mt.pair_suites ref = ref []
let test_id = ref 0

let eq loc x y =
  incr test_id ;
  suites :=
    (loc ^ " id " ^ string_of_int !test_id, fun _ -> Mt.Eq (x, y)) :: !suites

type shape = Circle of int | Rectangle of int * int

let myShape = Circle 10

let area =
  match myShape with
  | Circle r -> float_of_int (r * r) *. 3.14
  | Rectangle (w, h) -> float_of_int (w * h)

;;
eq __LOC__ area 314.
;;
Mt.from_pair_suites __MODULE__ !suites
